From e65eb3c0c6068dded75018c500b2338b8a41e42e Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 16 Dec 2023 13:25:53 +0000
Subject: [PATCH 1/1] sudo: Extremely basic statelessness support

Just gets vendor config paths and admin overriding working for now. Needs
more thought w.r.t visudo and such.
---
 src/sudo/mod.rs | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/src/sudo/mod.rs b/src/sudo/mod.rs
index e1e686a..10a3ff3 100644
--- a/src/sudo/mod.rs
+++ b/src/sudo/mod.rs
@@ -23,13 +23,27 @@ mod pipeline;
 const VERSION: &str = std::env!("CARGO_PKG_VERSION");
 
 fn candidate_sudoers_file() -> &'static Path {
+
+    // admin paths
     let pb_rs: &'static Path = Path::new("/etc/sudoers-rs");
+    let pb: &'static Path = Path::new("/etc/sudoers");
+
+    // vendor paths
+    let ven_rs: &'static Path = Path::new("/usr/share/defaults/etc/sudo/sudoers-rs");
+    let ven: &'static Path = Path::new("/usr/share/defaults/etc/sudo/sudoers");
+
     if pb_rs.exists() {
         dev_info!("Running with /etc/sudoers-rs file");
         pb_rs
-    } else {
+    } else if pb.exists()  {
         dev_info!("Running with /etc/sudoers file");
-        Path::new("/etc/sudoers")
+        pb
+    } else if ven_rs.exists() {
+        dev_info!("Running with /usr/share/defaults/etc/sudo/sudoers-rs file");
+        ven_rs
+    } else {
+        dev_info!("Running with /usr/share/defaults/etc/sudo/sudoers file");
+        ven
     }
 }
 
-- 
2.43.0

